### ISSUE TITLE
How to not auto-assign null to nullable properties?

### ISSUE THREAD

**Issue by eigan** [CuyZ/Valinor#113]
I want to differentiate between user not supplying a property, and supplying `null`.

```php
class Payload {
    // Must always be provided
    public int $id;
    
    // Can be null, or string,
    public string|null $note;
}
``` 

The source `['id' => 1]` will set `Payload::$note` to `null`. But user
did not supply `null` as value. I want to handle this a MappingError and ask
user to supply either null or string. Is this possible somehow?

**Comment by romm** [comment]
Hi @eigan, thanks for the report!

It looks like a valid usecase indeed; I'd link it to #3 which is next on my todo list.

**Comment by romm** [comment]
Hi @eigan, #159 should fix this. Can you take a look and see if this is solved on your side?

**Comment by romm** [comment]
Fixed with #159.
